main/autogen/subclasses.cc

```diff
@@ -92,34 +95,64 @@ optional<Subclasses::SubclassInfo> Subclasses::descendantsOf(const Subclasses::M
-                                                 childMap["Opus::Risk::Model::Mixins::RiskSafeMachine"].entries.end());
+void Subclasses::patchChildMap(const core::GlobalState &gs, Subclasses::Map &childMap) {
+    auto riskSafeMachineRef = getConstantRef(gs, "Opus::Risk::Model::Mixins::RiskSafeMachine");
+    childMap[getConstantRef(gs, "Opus::SafeMachine")].entries.insert(childMap[riskSafeMachineRef].entries.begin(),
+                                                                     childMap[riskSafeMachineRef].entries.end());
 }
 
-vector<string> Subclasses::serializeSubclassMap(const Subclasses::Map &descendantsMap,
-                                                const vector<string> &parentNames) {
+vector<string> Subclasses::serializeSubclassMap(const core::GlobalState &gs, const Subclasses::Map &descendantsMap,
+                                                const vector<core::SymbolRef> &parentNames, const bool showPaths) {
     vector<string> descendantsMapSerialized;
+    const auto classFormatString = showPaths ? " class {} {}" : " class {}";
+    const auto moduleFormatString = showPaths ? " module {} {}" : " module {}";
 
-    for (const string &parentName : parentNames) {
-        auto fnd = descendantsMap.find(parentName);
-        if (fnd == descendantsMap.end()) {
+    for (const auto &parentRef : parentNames) {
+        auto fnd = descendantsMap.find(parentRef);
+        if (fnd == descendantsMap.end() || !parentRef.exists()) {
             continue;
         }
         const Subclasses::SubclassInfo &children = fnd->second;
 
+        string parentName =
+            fmt::format("{}", fmt::map_join(symbolName(gs, parentRef),
+                                            "::", [&gs](const core::NameRef &nm) -> string { return nm.show(gs); }));
+
         auto type = children.classKind == ClassKind::Class ? "class" : "module";
         descendantsMapSerialized.emplace_back(fmt::format("{} {}", type, parentName));
 
         auto subclassesStart = descendantsMapSerialized.size();
-        for (const auto &[name, type] : children.entries) {
+        for (const auto &sym : children.entries) {
+            string_view path = sym.loc(gs).file().data(gs).path();
+            // Strip initial "./" from path
+            if (path.find("./") == 0)
+                path = path.substr(2);
+
+            string childName =
+                fmt::format("{}", fmt::map_join(symbolName(gs, sym), "::", [&gs](const core::NameRef &nm) -> string {
+                                return nm.show(gs);
+                            }));
+
             // Ignore Modules
-            if (type == autogen::Definition::Type::Class) {
-                descendantsMapSerialized.emplace_back(fmt::format(" class {}", name));
+            if (sym.asClassOrModuleRef().data(gs)->isClass()) {
+                // Note: fmt ignores excess arguments
+                descendantsMapSerialized.emplace_back(fmt::format(classFormatString, childName, path));
```

Comment:
I don't know if this is intentional--in fact I think a future change to the fmt library will change this behavior.

Let's do something like this:

```suggestion
                descendantsMapSerialized.emplace_back(fmt::format(" class {} {}", childName, showPaths ? path : ""));
```